api/test/gateways/UHW/UHWFetchDocuments.test.js

```diff
@@ -0,0 +1,66 @@
+const UHWFetchDocuments = require('../../../lib/gateways/UHW/UHWFetchDocuments');
+
+describe('UHWFetchDocumentsGateway', () => {
+  let buildDocument;
+  let db;
+
+  const createGateway = (records, throwsError) => {
+    buildDocument = jest.fn(({}) => {
+      return {};
+    });
+
+    db = {
+      request: jest.fn(async () => {
+        if (throwsError) {
+          throw new Error('Database error');
+        }
+        return records;
+      })
+    };
+
+    return UHWFetchDocuments({
+      buildDocument,
+      db
+    });
+  };
+
+  it('queries the database with the id if the query contains ID', async () => {
+    const gateway = createGateway([]);
+    const id = '123';
+    const paramMatcher = expect.arrayContaining([
+      expect.objectContaining({ value: '123' })
+    ]);
+
+    await gateway.execute(id);
+
+    expect(db.request).toHaveBeenCalledWith(expect.anything(), paramMatcher);
+  });
+
+  it('does not query the db for id if the query does not contain id', async () => {
+    const gateway = createGateway([]);
+
+    const documents = await gateway.execute();
+
+    expect(db.request).toHaveBeenCalledTimes(0);
+    expect(documents.length).toBe(0);
+  });
+
+  it('builds a document', async () => {
+    const document = { DocNo: '1231' };
+    const gateway = createGateway([document]);
+
+    const documents = await gateway.execute({});
+    const paramMatcher = expect.objectContaining({ id: '1231' });
+    expect(buildDocument).toHaveBeenCalledWith(paramMatcher);
+    expect(documents.length).toBe(1);
```

Comment:
Is this line necessary?

Reply:
Not sure which line you're talking about but if it's 55, the I think so? It checks that it only creates 1 document and not more. What are the downsides of having it?

Reply:
Line 54 checks that it is called once, line 55 kinda tests that our fake implementation returns something